vendor/processors/stripe_processor.py

```diff
@@ -341,6 +349,10 @@ def build_price(self, offer, price):
         if 'stripe' not in offer.meta or 'product_id' not in offer.meta['stripe']:
             raise TypeError(f"Price cannot be created without a product_id on offer.meta['stripe'] field")
 
+        if isinstance(price, (int, float)):
+            raise TypeError(f"Price cannot be created without a valid vendor price object added to this offer")
```

Comment:
An offer many not have an price model object, because it can use the msrp from the product which is defined in the meta field in the product.